client/src/utils/api.js

```diff
@@ -174,3 +175,110 @@ export const deleteResource = (id: string): ApiResponse<null> => {
       },
     );
 };
+
+export const addCategory = (category: Category): ApiResponse<Category> =>
+  instance
+    .post('/api/admin/categories', category, {
+      headers: {
+        token: localStorage.getItem('token'),
+      },
+    })
+    .then(
+      (res) => res.data,
+      (err) => {
+        console.error(err);
+        return null;
+      },
+    );
+
+export const editCategory = (
+  id: string,
+  category: Category,
+): ApiResponse<Category> => {
+  const requestExtension = `/api/admin/categories/${id}`;
+  return instance
+    .put(requestExtension, category, {
+      headers: {
+        token: localStorage.getItem('token'),
+      },
+    })
+    .then(
+      (res) => res.data,
+      (err) => {
+        console.error(err);
+        return null;
+      },
+    );
+};
+
+export const deleteCategory = (id: string): ApiResponse<null> => {
+  const requestExtension = `/api/admin/categories/${id}`;
+  return instance
+    .delete(requestExtension, {
+      headers: {
+        token: localStorage.getItem('token'),
+      },
+    })
+    .then(
+      (res) => res.data,
+      (err) => {
+        console.error(err);
+        return null;
+      },
+    );
+};
+
+export const addSubcategory = (
+  id: string,
+  category: Category
```

Comment:
Since we're just using `req.body.name` in the endpoint, we can just make this parameter `name: string`, and make the post call ```.post(`/api/admin/subcategories/${id}`, { name }```